labgrid/driver/networkinterfacedriver.py

```diff
@@ -113,3 +124,64 @@ def get_access_points(self, scan=None):
         if the last scan is more than 30 seconds old.
         """
         return self.proxy.get_access_points(self.iface.ifname, scan)
+
+    @Driver.check_active
+    @contextlib.contextmanager
+    def forward_local(self, remotehost, remoteport, localport=None):
+        """Forward a local port to an address and port via the exporter.
+
+        A context manager that keeps the local port forwarded as long as the
+        context remains valid. A connection can be made to the returned port on
+        localhost and it will be forwarded to the address and port.
+
+        If localport is not set, a random free port will be selected.
+
+        usage:
+            with netif.forward_local('127.0.0.1', 8080, localport=1234) as localport:
+                # Use localhost:1234 here to connect to port 8080 on
+                # the exporter
+
+            with netif.forward_local('192.168.1.2', 8080) as localport:
+                # Use localhost:localport here to connect to port 8080 on
+                # 192.168.1.2
+
+        returns:
+            localport
+        """
+        if not self.ssh:
+            raise ExecutionError("Resource is not on an exporter")
+
+        localport = self.ssh.add_port_forward(remotehost, remoteport, localport)
+        try:
+            yield localport
+        finally:
+            self.ssh.remove_port_forward(remotehost, remoteport)
+
+    @Driver.check_active
+    @contextlib.contextmanager
+    def forward_remote(self, remoteport, localport, remotebind=None):
+        """Forward a remote port on the exporter to a local port.
+
+        A context manager that keeps a remote port forwarded to a local port as
+        long as the context remains valid. A connection can be made to the
+        remote port on the target device will be forwarded to the returned local
+        port on localhost.
+
+        Note that the remote socket is not *bound* to any specific IP by
+        default, making it reachable by the target. Also, 'GatewayPorts
+        clientspecified' needs to be configured in the remote host's
+        sshd_config.
+
+        usage:
+            with netif.forward_remote(8080, 8081) as localport:
+                # Connections to port 8080 on the exporter will be redirected to
+                # localhost:8081
+        """
+        if not self.ssh:
+            raise ExecutionError("Resource is not on an exporter")
+
+        self.ssh.add_remote_port_forward(remoteport, localport, remotebind)
+        try:
+            yield
+        finally:
+            self.ssh.remove_remote_port_forward(remoteport, localport, remotebind)
```

Comment:
I've changed `remove_remote_port_forward()` to accept None to select the default behavior.

Reply:
Why do you use `None` instead of `"*"` as the default for the remote binds kwargs?
Why is it sometimes called `remotebind` and sometimes `remote_bind`?

Reply:
Okay, thanks.

Reply:
I've renamed the parameters to `remote_host`, `remote_port`, `local_port` and `remote_bind`.